### PR TITLE
Add option to build the SIMDjson library

### PR DESCRIPTION
**Pull request by urysegal**
See https://github.com/lemire/simdjson

### CODE REVIEW

File: SonarExternal.cmake [hunk 1]
```diff
@@ -2006,3 +2006,28 @@ function(build_sasl)
...
+      -DBUILD_SHARED_LIBS=NO
+      -DCMAKE_CXX_COMPILER_LAUNCHER=${CMAKE_CXX_COMPILER_LAUNCHER}
+      -DCMAKE_C_COMPILER_LAUNCHER=${CMAKE_C_COMPILER_LAUNCHER}
+      -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+      -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+      -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+    )
+  external_project_dirs(simdjson install_dir)
+  add_library(simdjson::lib INTERFACE IMPORTED)
+  target_include_external_directory(simdjson::lib simdjson install_dir include)
```
Comment (yhagerjs) [inline SonarExternal.cmake]: missing `set_target_properties(simdjson::lib IMPORTED_LOCATION ${simdjson_install_dir}/lib/libsimdjson.a)` (assuming that's the correct path).

File: SonarExternal.cmake [hunk 2]
```diff
@@ -2006,3 +2006,28 @@ function(build_sasl)
...
+  ExternalProject_Add(simdjson
+    URL https://github.com/lemire/simdjson/archive/v${SIMDJSON_VERSION}.tar.gz
+    DOWNLOAD_NO_PROGRESS ON
+    CMAKE_ARGS
+      -DBUILD_SHARED_LIBS=NO
+      -DCMAKE_CXX_COMPILER_LAUNCHER=${CMAKE_CXX_COMPILER_LAUNCHER}
+      -DCMAKE_C_COMPILER_LAUNCHER=${CMAKE_C_COMPILER_LAUNCHER}
+      -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+      -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+      -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
```
Comment (yhagerjs) [inline SonarExternal.cmake]: missing `BUILD_BYPRODUCTS <INSTALL_DIR>/lib/libsimdjson.a` (assuming that's the correct path)

File: SonarExternal.cmake (hunk 1 above)
Comment (yhagerjs) [inline SonarExternal.cmake]: `target_include_external_directory` is deprecated. Please use `include_external_directories(TARGET simdjson::lib DIRECTORIES ${simdjson_install_dir}/include)` instead.

File: SonarExternal.cmake [hunk 3]
```diff
@@ -2006,3 +2006,28 @@ function(build_sasl)
...
+    )
+  external_project_dirs(simdjson install_dir)
+  add_library(simdjson::lib INTERFACE IMPORTED)
+  target_include_external_directory(simdjson::lib simdjson install_dir include)
+  set_property(TARGET simdjson::lib
+    PROPERTY INTERFACE_COMPILE_DEFINITIONS
```
Comment (yhagerjs) [inline SonarExternal.cmake]: what's the value of the property you are setting?

File: SonarExternal.cmake (hunk 1 above)
Comment (urysegal) [reply SonarExternal.cmake]: There is no .a file - it's a pure .h file library

File: SonarExternal.cmake (hunk 2 above)
Comment (urysegal) [reply SonarExternal.cmake]: There is no .a file - it's a pure .h file library

File: SonarExternal.cmake (hunk 1 above)
Comment (yhagerjs) [reply SonarExternal.cmake]: Then change the target name to be `simdjson::header-only`

File: SonarExternal.cmake (hunk 3 above)
Comment (urysegal) [reply SonarExternal.cmake]: I removed this line.

File: SonarExternal.cmake (hunk 1 above)
Comment (urysegal) [reply SonarExternal.cmake]: done

Comment (urysegal) [reply SonarExternal.cmake]: ok done

**Pull request merged**